lib/_http_outgoing.js

```diff
@@ -426,6 +430,37 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return (this._headers ? Object.keys(this._headers) : []);
+};
+
+
+// Returns a shallow copy of the current outgoing headers.
```

Comment:
Well, it works, "some people may find it useful", and we should either warn people that they should not do it, because the feature may go away, or doc it as a feature, so they know they can rely on it. We have difficulty historically in defining what is or is not the node API, I would prefer we be start being more clear about this. cc: @gibfahn